test/e2e_node/image_list.go

```diff
@@ -151,27 +156,61 @@ func PrePullAllImages() error {
 	}
 	images := framework.ImageWhiteList.List()
 	klog.V(4).Infof("Pre-pulling images with %s %+v", puller.Name(), images)
-	for _, image := range images {
-		var (
-			err    error
-			output []byte
-		)
-		for i := 0; i < maxImagePullRetries; i++ {
-			if i > 0 {
-				time.Sleep(imagePullRetryDelay)
-			}
-			if output, err = puller.Pull(image); err == nil {
-				break
+
+	imageCh := make(chan int, len(images))
+	for i := range images {
```

Comment:
there is a deathlock in the previous implementation, I did fix it and all tests have passed now. PTAL @mattjmcnaughton

Reply:
Can you share more around what the deadlock was @lsytj0413 ? Would also love to know more around how we are confident this implementation avoids the deadlock. Thanks :)

Reply:
Used variable imageCh in the for range sentence at previous implementation，it should be images。And read from imageCh will be deathlock。